lib/engines/hcktest/tests.rb

```diff
@@ -389,22 +433,19 @@ def reset_clients_to_ready_state
       @support&.reset_to_ready_state
     end
 
-    def new_done
-      list_tests
-      done_tests - @last_done
-    end
-
     def apply_filters
       @logger.info('Applying filters on finished tests')
       @tools.apply_project_filters(@tag)
       sleep APPLYING_FILTERS_INTERVAL
     end
 
-    def check_new_finished_tests
-      return unless new_done.any?
+    def check_new_finished_results
+      # Update tests results to get the latest status of the tests
+      update_tests_results
```

Comment:
Let's call this method at `handle_test_running` as it computes not only `@new_done_test_results` but also `@tests_results`, which affects more than the following `handle_finished_test_results` call.